components/camel-dynamic-router/src/main/java/org/apache/camel/component/dynamicrouter/DynamicRouterEndpoint.java

```diff
@@ -140,30 +149,87 @@ protected void doInit() throws Exception {
+            DynamicRouterMulticastProcessor processor = processorFactorySupplier.get()
+                    .getInstance("DynamicRouterMulticastProcessor-" + configuration.getChannel(), camelContext, null,
+                            configuration.getRecipientMode(),
+                            configuration.isWarnDroppedMessage(), filterProcessorFactorySupplier, producerCache,
+                            aggregationStrategy, configuration.isParallelProcessing(),
+                            determineExecutorService(camelContext), configuration.isShutdownExecutorService(),
+                            configuration.isStreaming(), configuration.isStopOnException(), timeout,
+                            determineOnPrepare(camelContext), configuration.isShareUnitOfWork(),
+                            configuration.isParallelAggregate());
+            processor.setErrorHandler(errorHandler);
+            processor.setAggregateExecutorService(aggregateExecutorService);
+            processor.setIgnoreInvalidEndpoints(configuration.isIgnoreInvalidEndpoints());
+            processor.setId(getId());
+            processor.setRouteId(routeId);
+            ServiceHelper.startService(aggregationStrategy, producerCache, processor);
             component.addRoutingProcessor(configuration.getChannel(), processor);
         }
     }
 
+    protected ExecutorService determineExecutorService(CamelContext camelContext) {
+        ExecutorService executorService = null;
+        if (ObjectHelper.isNotEmpty(configuration.getExecutorService())) {
+            executorService = camelContext.getExecutorServiceManager()
+                    .newThreadPool(this, "@RecipientList", configuration.getExecutorService());
+        }
+        if (configuration.isParallelProcessing() && configuration.getExecutorService() == null) {
+            // we are running in parallel, so we need a thread pool
+            executorService = camelContext.getExecutorServiceManager()
+                    .newDefaultThreadPool(this, "@RecipientList");
+        }
+        return executorService;
+    }
+
+    protected AggregationStrategy determineAggregationStrategy(CamelContext camelContext) {
+        AggregationStrategy aggregationStrategy = new UseLatestAggregationStrategy();
+        if (ObjectHelper.isNotEmpty(configuration.getAggregationStrategy())) {
+            aggregationStrategy = CamelContextHelper.mandatoryLookup(camelContext,
+                    configuration.getAggregationStrategy(), AggregationStrategy.class);
+        }
+        return aggregationStrategy;
+    }
+
+    protected Processor determineOnPrepare(CamelContext camelContext) {
+        Processor processor = exchange -> {
+        };
+        if (ObjectHelper.isNotEmpty(configuration.getOnPrepare())) {
+            processor = CamelContextHelper.mandatoryLookup(camelContext, configuration.getOnPrepare(), Processor.class);
+        }
+        return processor;
```

Comment:
Maybe return a static processor to avoid the lambda/instantiation overhead. Something like:


```
  
        if (ObjectHelper.isNotEmpty(configuration.getOnPrepare())) {
                return CamelContextHelper.mandatoryLookup(camelContext, configuration.getOnPrepare(), Processor.class);
        }
        return SOME_STATIC_EMPTY_PROCESSOR;
```

Reply:
Thanks.  I will change it as requested.